python/feathub/processors/flink/table_builder/tests/test_flink_sql_feature_view.py

```diff
@@ -0,0 +1,322 @@
+#  Copyright 2022 The Feathub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import typing
+from abc import ABC
+from datetime import timedelta
+
+import pandas as pd
+
+from feathub.common import types
+from feathub.common.exceptions import FeathubException
+from feathub.common.test_utils import to_epoch_millis
+from feathub.common.types import Int64
+from feathub.feature_tables.sources.datagen_source import DataGenSource
+from feathub.feature_views.feature import Feature
+from feathub.feature_views.feature_view import FeatureView
+from feathub.feature_views.sliding_feature_view import (
+    SlidingFeatureView,
+    ENABLE_EMPTY_WINDOW_OUTPUT_CONFIG,
+    SKIP_SAME_WINDOW_OUTPUT_CONFIG,
+)
+from feathub.feature_views.sql_feature_view import SqlFeatureView
+from feathub.feature_views.transforms.sliding_window_transform import (
+    SlidingWindowTransform,
+)
+from feathub.processors.flink.flink_processor import FlinkProcessor
+from feathub.registries.local_registry import LocalRegistry
+from feathub.table.schema import Schema
+from feathub.tests.feathub_it_test_base import FeathubITTestBase
+
+
+# TODO: Supports the cases that using a SqlFeatureView as a direct or
+#  indirect source of another SqlFeatureView.
+class FlinkSqlFeatureViewITTest(ABC, FeathubITTestBase):
+    def setUp(self) -> None:
+        FeathubITTestBase.setUp(self)
+        self.source = self.create_file_source(
+            self.input_data.copy(), name="__TEST_SOURCE__"
+        )
+        self.client.build_features([self.source])
+
+    def tearDown(self) -> None:
+        FeathubITTestBase.tearDown(self)
+        t_env = typing.cast(
+            FlinkProcessor, self.client.processor
+        ).flink_table_builder.t_env
+        for view_name in t_env.list_temporary_views():
+            t_env.drop_temporary_view(view_name)
+        typing.cast(LocalRegistry, self.client.registry).tables.clear()
```

Comment:
Should this be moved to `FeathubITTestBase#tearDown`?

It seems more readable to only do it when registry is an instance of `LocalRegistry`. And add public API like `LocalRegistry#clear()`.